qiskit_nature/problems/second_quantization/molecular/fermionic_op_builder.py

```diff
@@ -0,0 +1,105 @@
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Fermionic operator builder. """
+
+import itertools
+
+import numpy as np
+
+from qiskit_nature.drivers.qmolecule import QMolecule
+from qiskit_nature.operators import FermionicOp
+
+
+def build_fermionic_op(q_molecule: QMolecule) -> FermionicOp:
+    """
+    Builds a fermionic operator based on a QMolecule object.
+
+    Args:
+        q_molecule (QMolecule): QMolecule instance with 1- and/or 2-body integrals.
+
+    Returns:
+        FermionicOp: FermionicOp built from a QMolecule object.
+    """
+
+    one_body_ints = q_molecule.one_body_integrals
+    two_body_ints = q_molecule.two_body_integrals
+
+    return build_ferm_op_from_ints(one_body_ints, two_body_ints)
+
+
+def build_ferm_op_from_ints(one_body_integrals: np.ndarray,
+                            two_body_integrals: np.ndarray = None) -> FermionicOp:
+    """
+    Builds a fermionic operator based on 1- and/or 2-body integrals.
+    This method requires the integrals stored in the '*chemist*' notation
+             h2(i,j,k,l) --> adag_i adag_k a_l a_j
+    and the integral values are used for the coefficients of the second-quantized
+    Hamiltonian that is built. The integrals input here should be in block spin
+    format and also have indexes reordered as follows 'ijkl->ljik'
+    There is another popular notation, the '*physicist*' notation
+             h2(i,j,k,l) --> adag_i adag_j a_k a_l
+    If you are using the '*physicist*' notation, you need to convert it to
+    the '*chemist*' notation. E.g. h2=numpy.einsum('ikmj->ijkm', h2)
+    The :class:`~qiskit_nature.QMolecule` class has
+    :attr:`~qiskit_nature.QMolecule.one_body_integrals` and
+    :attr:`~qiskit_nature.QMolecule.two_body_integrals` properties that can be
+    directly supplied to the `h1` and `h2` parameters here respectively.
+
+    Args:
+        one_body_integrals (numpy.ndarray): One-body integrals stored in the chemist notation.
+        two_body_integrals (numpy.ndarray): Two-body integrals stored in the chemist notation.
+
+    Returns:
+        FermionicOp: FermionicOp built from 1- and/or 2-body integrals.
+    """
+
+    fermionic_op = FermionicOp('I' * len(one_body_integrals))
+    fermionic_op = _fill_ferm_op_one_body_ints(fermionic_op, one_body_integrals)
+    if two_body_integrals is not None:
+        fermionic_op = _fill_ferm_op_two_body_ints(fermionic_op, two_body_integrals)
+
+    fermionic_op = fermionic_op.reduce()
+
+    return fermionic_op
+
+
+def _fill_ferm_op_one_body_ints(fermionic_op: FermionicOp,
```

Comment:
Not that I want to get too hung up over names but would something like `_add_one_body_terms` and `_add_two_body_terms` be "better". 

I am wondering on the efficiency of building the FermionicOp out this way. Would it be more efficient just to build the list of labels+coeffs and then make the FermionicOp from that list in a single go? Again I don't want to get hung up over early optimization though...

Reply:
I like names you suggested!

Reply:
I am not sure. Since operations on the FermionicOp like adding a new term make a new one with the combined set of terms, I was thinking that building out the terms as a complete list and contructing a FermionicOp once with the complete list would be more efficient.

Reply:
Ok, now I can see how `+` and `@` are overloaded and fully see what you mean.